tools/kagami/src/main.rs

```diff
@@ -0,0 +1,298 @@
+    let app = App::new("Kagami")
+        .version("0.1")
+        .author("Iroha development team.")
+        .about("Generator for data used in Iroha.")
+        .arg(
+            Arg::with_name("seed")
+                .long("seed")
+                .value_name("seed")
+                .help("Sets a seed for random number generator. Should be used separately from `private_key`.")
+                .required(false)
+                .takes_value(true)
+        )
+        .arg(
+            Arg::with_name("private-key")
+                .long("private-key")
+                .value_name("private_key")
+                .help("Sets a private key. Should be used separately from `seed`.")
+                .required(false)
+                .takes_value(true)
+        )
+        .arg(
+            Arg::with_name("algorithm")
+                .long("algorithm")
+                .value_name("algorithm")
+                .help("Function used to generate the key pair.")
+                .takes_value(true)
+                .possible_value(iroha_crypto::ED_25519)
+                .possible_value(iroha_crypto::SECP_256_K1)
+                .possible_value(iroha_crypto::BLS_NORMAL)
+                .possible_value(iroha_crypto::BLS_SMALL)
+                .default_value(DEFAULT_ALGORITHM)
+        )
+        .arg(
+            Arg::with_name("json")
+            .long("json")
+            .help("If specified the output will be formatted as json.")
+            .takes_value(false)
+            .multiple(false)
+        )
+        .arg(
+            Arg::with_name("genesis")
+                .long("genesis")
+                .short("g")
+                .help("If specified, print the Genesis")
+                .takes_value(false)
+                .multiple(false)
+        )
+        .arg(
+            Arg::with_name("schema")
+                .long("schema")
+                .short("s")
+                .help("If specified, print Schema")
+                .takes_value(false)
+                .multiple(false)
+        )
+        .arg(
+            Arg::with_name("docs")
+                .long("docs")
+                .short("d")
+                .help("If specified, print configuration docs")
+                .takes_value(false)
+                .multiple(false)
+        )
+        .group(
+            ArgGroup::with_name("other_gen_options")
+                .args(&["docs", "schema", "genesis"])
+                .required(false)
+                .multiple(false)
+        )
+        .group(
+            ArgGroup::with_name("key_gen_options")
+                .args(&["seed", "private-key"])
+                .required(false)
+                .multiple(false)
+        ).get_matches();
+    app
+}
+
+mod crypto {
+
+    use color_eyre::eyre::{eyre, WrapErr as _};
+    use iroha_crypto::{Algorithm, KeyGenConfiguration, KeyPair, PrivateKey};
+
+    pub fn key_pair(matches: &clap::ArgMatches) -> color_eyre::Result<KeyPair> {
```

Comment:
There is no checking for both `seed` and `private_key` options enabled here. I think you should do checking or rewrite help message and mention that if both specified then `private_key` option will be ignored

Reply:
[ArgGroup::multiple(false)](https://docs.rs/clap/latest/clap/struct.ArgGroup.html#method.multiple) ensures one of them are accepted